### PR TITLE
Content Pipeline: Move has_lesson_plan? to the script DSL

### PR DESCRIPTION
**Pull request by islemaster**
Replaces this hard-coded `Script.has_lesson_plan?` helper...

https://github.com/code-dot-org/code-dot-org/blob/f576ccd93490c8e71775952563cc3b4aa1b11bdc/dashboard/app/models/script.rb#L520-L522

...with a new `has_lesson_plan` field in the Script DSL and related support.  This eliminates one of the documented ["Steps to make a lesson plan:"](https://docs.google.com/document/d/1v8BxwLPc_547L-nmgnIUs8jdVb1GaQyCECmZx_YuPs4/edit#bookmark=id.xx7ohlu0aa10)

> * Update script.rb via github to tell LB it has a lesson.
>   * Example [PR](https://github.com/code-dot-org/code-dot-org/pull/18871/files)
>   * Josh and Baker make these PRs via github, others ask an engineer to do it

Now, `has_lesson_plan` is a serialized attribute on `Script`, which implicitly creates a `has_lesson_plan?` helper method replacing the old one which has now been removed.

I've added a checkbox to the script edit view on levelbuilder, so our levelbuidlers can control this attribute without needing to interact with git or script files directly.

image

All existing scripts that had lesson plans have been updated, at time of writing.

I've added test coverage for `Script`, `ScriptDSL` and `ScriptsController` around the new `has_lesson_plan` attribute.

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -517,10 +518,6 @@ def cs_in_a?
     name.match(Regexp.union('algebra', 'Algebra'))
   end
 
-  def has_lesson_plan?
-    k5_course? || k5_draft_course? || %w(msm algebra algebraa algebrab cspunit1 cspunit2 cspunit3 cspunit4 cspunit5 cspunit6 csp1 csp2 csp3 csp4 csp5 csp6 csppostap cspoptional csp3-research-mxghyt csd1 csd2 csd3 csd4 csd5 csd6 csp-ap text-compression netsim pixelation frequency_analysis vigenere applab-intro csp-explore csp-create).include?(name)
```
Comment (islemaster) [inline dashboard/app/models/script.rb:521]: Every script referenced in this conditional was updated, except for `msm.script` which I can't find - I don't think it exists anymore.

**Comment by islemaster** [comment]
Working on test failure:
```
Failure:
test_should_show_new_style_unplugged_level_with_PDF_link(Minitest::Result) [/home/circleci/project/dashboard/test/controllers/script_levels_controller_test.rb:581]:
Expected exactly 2 elements matching ".pdf-button", found 1..
Expected: 2
  Actual: 1
```

**Comment by caleybrock** [comment]
looking

**Comment by islemaster** [comment]
I think that removing the line from the .script file will set `has_lesson_plan` back to its default value (false), very similar to how all of our scripts default to `hidden` unless `hidden false` is explicitly set in the .script file.  I thought this was true for all DSL attributes, but I may be wrong.  I think I could cover this with a `Script` model test that simulates reseeding an existing script?

**Comment by Hamms** [comment]
oh, good point about the hidden attribute. Given that, I don't feel a strong need for a test

**Pull request merged**